apps/src/panels/PanelsView.tsx

```diff
@@ -0,0 +1,164 @@
+// Panels
+//
+// This is a React client for a panels level.  Note that this is
+// only used for levels that use Lab2.
+
+import React, {useEffect, useLayoutEffect, useState} from 'react';
+import {useSelector} from 'react-redux';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {
+  sendSuccessReport,
+  navigateToNextLevel,
+} from '@cdo/apps/code-studio/progressRedux';
+import {LabState} from '@cdo/apps/lab2/lab2Redux';
+import {PanelsLevelData} from '@cdo/apps/lab2/types';
+import panelsLocale from './locale';
+import styles from './panels.module.scss';
+import classNames from 'classnames';
+import EnhancedSafeMarkdown from '@cdo/apps/templates/EnhancedSafeMarkdown';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+function useWindowSize() {
+  const [size, setSize] = useState([
+    document.documentElement.clientWidth,
+    document.documentElement.clientHeight,
+  ]);
+  useLayoutEffect(() => {
+    function updateSize() {
+      const width = document.documentElement.clientWidth;
+      const height = document.documentElement.clientHeight;
+      setSize([width, height]);
+    }
+    window.addEventListener('resize', updateSize);
+    updateSize();
+    return () => window.removeEventListener('resize', updateSize);
+  }, []);
+  return size;
+}
+
+const PanelsView: React.FunctionComponent = () => {
+  const dispatch = useAppDispatch();
+  const levelData = useSelector(
+    (state: {lab: LabState}) => state.lab.levelProperties?.levelData
+  );
+  const currentAppName = useSelector(
+    (state: {lab: LabState}) => state.lab.levelProperties?.appName
+  );
```

Comment:
Toss the `state` variable and just destructure it:

```
({lab}: {lab: LabState}) => lab.levelProperties?.levelData
```

Reply:
Interesting.  For now, I've just tossed the `state` variable.